src/cargo/core/shell.rs

```diff
@@ -340,6 +356,63 @@ impl Shell {
         }
     }
 
+    pub fn out_hyperlink<D: fmt::Display>(&self, url: D) -> Hyperlink<D> {
+        let supports_hyperlinks = match &self.output {
+            ShellOut::Write(_) => false,
+            ShellOut::Stream {
+                stdout, hyperlinks, ..
+            } => stdout.current_choice() == anstream::ColorChoice::AlwaysAnsi && *hyperlinks,
+        };
+        if supports_hyperlinks {
+            Hyperlink { url: Some(url) }
+        } else {
+            Hyperlink { url: None }
+        }
+    }
+
+    pub fn err_hyperlink<D: fmt::Display>(&self, url: D) -> Hyperlink<D> {
+        let supports_hyperlinks = match &self.output {
+            ShellOut::Write(_) => false,
+            ShellOut::Stream {
+                stderr, hyperlinks, ..
+            } => stderr.current_choice() == anstream::ColorChoice::AlwaysAnsi && *hyperlinks,
+        };
+        if supports_hyperlinks {
+            Hyperlink { url: Some(url) }
+        } else {
+            Hyperlink { url: None }
+        }
+    }
+
+    pub fn out_file_hyperlink(&mut self, path: &std::path::Path) -> Hyperlink<url::Url> {
+        let url = self.file_hyperlink(path);
+        url.map(|u| self.out_hyperlink(u)).unwrap_or_default()
+    }
+
+    pub fn err_file_hyperlink(&mut self, path: &std::path::Path) -> Hyperlink<url::Url> {
```

Comment:
Need to test that this actually works (unsure if its a bug or my terminal is too old)

Reply:
Ah, apparently I missed [this note from burntsushi](https://github.com/BurntSushi/ripgrep/discussions/2611)

> (NOTE: If you're using tmux, its latest release doesn't support hyperlinks. You'll need to build tmux from source and then add set-option -sa terminal-features ",*:hyperlinks" to your tmux configuration to make it work.)

Reply:
Hmm, just tried switching from tmux to Zellij.  It now highlights the links but they don't work.  With a raw terminal in Wezterm, the links work.